### ISSUE TITLE
Explosive bullets do not deal splash damage

### ISSUE THREAD

**Issue by Dubbington** [DonBruce64/MinecraftTransportSimulator#1561]
About time I ticket this

Explosive bullets do not deal splash damage to entities unless the following two occur:
- Bullet directly hits an entity, resulting in actual explosive splash damage
- Player firing explosive bullet happens to be wearing Thorns, dealing explosive damage to the player

This was tested on [Immersive_Vehicles-1.12.2-22.6.0-BETA.jar] with the bullet impact fix.

**Comment by boot2big** [comment]
Try adding a proximityFuse to the bullet? I could have sworn I had a similar issue and this at least mitigated it.

**Comment by DonBruce64** [comment]
No, he's correct.  Explosive bullets don't apply potion effects in their blast radius.  It's because the explosions are Minecraft explosions and don't know anything about potions.  If someone would come up with a formular to make explosions internal to IV, I could fix that.

**Comment by ktokto313** [comment]
Is this a ticket about explosion or about the potion effect of the explosion? We can reuse the code in minecraft but make it more flexible using variable, not hardcoded number. Something like flan mod did exactly that.

**Comment by DonBruce64** [comment]
Addressed in 22.18.0.
